internal/datastore/mysql/datastore.go

```diff
@@ -118,7 +118,27 @@ func newMySQLDatastore(ctx context.Context, uri string, options ...Option) (*Dat
 		return nil, errors.New("error in NewMySQLDatastore: connection URI for MySQL datastore must include `parseTime=true` as a query parameter; see https://spicedb.dev/d/parse-time-mysql for more details")
 	}
 
-	connector, err := mysql.MySQLDriver{}.OpenConnector(uri)
+	// Setup the credentials provider
+	var credentialsProvider datastore.CredentialsProvider
+	if config.credentialsProviderName != "" {
+		credentialsProvider, err = datastore.NewCredentialsProvider(ctx, config.credentialsProviderName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if credentialsProvider != nil {
+		// add a before connect callback to trigger the token retrieval from the credentials provider
+		err := parsedURI.Apply(mysql.BeforeConnect(func(ctx context.Context, config *mysql.Config) error {
+			config.User, config.Passwd, err = credentialsProvider.Get(ctx, config.Addr, config.User)
+			return err
+		}))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	connector, err := mysql.NewConnector(parsedURI)
```

Comment:
We create the connector using `mysql.NewConnector(parsedURI)` instead of `mysql.MySQLDriver{}.OpenConnector(uri)`, so we can preserve the `BeforeConnect` hook we added